Microsoft.NET.Build.Containers/ImageConfig.cs

```diff
@@ -29,53 +37,108 @@ internal ImageConfig(JsonNode config)
         _labels = GetLabels();
         _exposedPorts = GetExposedPorts();
         _environmentVariables = GetEnvironmentVariables();
+        _rootFsLayers = GetRootFileSystemLayers();
+        _architecture = GetArchitecture();
+        _os = GetOs();
+        _history = GetHistory();
     }
 
+    private List<HistoryEntry> GetHistory() => _config["history"]?.AsArray().Select(node => node.Deserialize<HistoryEntry>()!).ToList() ?? new List<HistoryEntry>();
+    private string GetOs() => _config["os"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'os' property.");
+    private string GetArchitecture() => _config["architecture"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'architecture' property.");
+
     /// <summary>
     /// Builds in additional configuration and returns updated image configuration in JSON format as string.
     /// </summary>
     internal string BuildConfig()
     {
-        if (_config["config"] is not JsonObject config)
+        var newConfig = new JsonObject();
+
+        if (_exposedPorts.Any())
         {
-            throw new InvalidOperationException("Base image configuration should contain a 'config' node.");
+            newConfig["ExposedPorts"] = CreatePortMap();
+        }
+        if (_labels.Any())
+        {
+            newConfig["Labels"] = CreateLabelMap();
+        }
+        if (_environmentVariables.Any())
+        {
+            newConfig["Env"] = CreateEnvironmentVariablesMapping();
         }
-
-        //update creation date
-        _config["created"] = DateTime.UtcNow;
-  
-        config["ExposedPorts"] = CreatePortMap();
-        config["Env"] = CreateEnvironmentVariablesMapping();
-        config["Labels"] = CreateLabelMap();
 
         if (_newWorkingDirectory is not null)
         {
-            config["WorkingDir"] = _newWorkingDirectory;
+            newConfig["WorkingDir"] = _newWorkingDirectory;
         }
 
         if (_newEntryPoint.HasValue)
         {
-            config["Entrypoint"] = ToJsonArray(_newEntryPoint.Value.ExecutableArgs);
+            newConfig["Entrypoint"] = ToJsonArray(_newEntryPoint.Value.ExecutableArgs);
 
             if (_newEntryPoint.Value.Args is null)
             {
-                config.Remove("Cmd");
+                newConfig.Remove("Cmd");
             }
             else
             {
-                config["Cmd"] = ToJsonArray(_newEntryPoint.Value.Args);
+                newConfig["Cmd"] = ToJsonArray(_newEntryPoint.Value.Args);
             }
         }
 
-        foreach (Layer l in _newLayers)
+        // add a history entry for ourselves so folks can map generated layers to the Dockerfile commands
+        _history.Add(new HistoryEntry(DateTime.UtcNow, $".NET SDK Container Tooling, version {Constants.Version}", null, null, null));
```

Comment:
Open for suggestions on what we should put as the `created_by` here. For Docker this is the line from the Dockerfile that ended up creating the Layer.